### PR TITLE
[CRT-5] Added basic data model

### PR DESCRIPTION
**Pull request by pierluca**
Data model for:
- Users
- Students (anonymized)
- Class
- Session
- Task
- Solution
- SolutionAnalysis

### CODE REVIEW

File: backend/prisma/seed.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        where: { id: name },
+        update: {},
+        create: { id: name, classId: klass.id },
+      });
+
+      console.log(["student", student]);
+
+      const solution = await prisma.solution.upsert({
+        // This is a hack to make sure we have 1 solution per student
+        where: { id: index + 1 },
```
Comment (Tyratox) [inline backend/prisma/seed.ts:74]: I guess there is no simple insert method? Or you you want it do be idempotent?

**Review by Tyratox — CHANGES_REQUESTED** [review]
CI is broken ;)

But looks very nice!!

File: backend/prisma/migrations/20241030010049_class_session_task_solution/migration.sql
[the remark this answers is not in the document]
Comment (pierluca) [reply backend/prisma/migrations/20241030010049_class_session_task_solution/migration.sql:1]: It's generated

File: backend/prisma/seed.ts (hunk 1 above)
Comment (pierluca) [reply backend/prisma/seed.ts:74]: I used upsert for idempotency, yes :-)

File: backend/prisma/seed.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  const klass = await prisma.class.upsert({
+    where: { id: 1 },
+    update: {},
+    create: {
+      name: "Class 1A // 2024-2025",
+      teacherId: teacher.id,
+    },
+  });
+
+  console.log(["class", klass]);
```
[the remark this answers is not in the document]
Comment (pierluca) [reply backend/prisma/seed.ts:27]: for debugging purposes. The seeding will be reduced to the bare minimum when we're done with the full backend implementation.

File: backend/prisma/seed.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    create: {
+      email: "alfred@example.com",
+      name: "Alfred",
+      type: "TEACHER",
+    },
+  });
+
+  console.log(["teacher", teacher]);
+
+  const klass = await prisma.class.upsert({
```
[the remark this answers is not in the document]
Comment (pierluca) [reply backend/prisma/seed.ts:18]: nope, `class` is a keyword :-)

File: backend/prisma/migrations/20241030010049_class_session_task_solution/migration.sql
[the remark this answers is not in the document]
Comment (pierluca) [reply backend/prisma/migrations/20241030010049_class_session_task_solution/migration.sql:1]: Generated

**Review by Tyratox — APPROVED** [review]

**Pull request merged**